third_party/move/move-vm/runtime/src/data_cache.rs

```diff
@@ -146,17 +174,52 @@ impl TransactionDataCache {
         let (layout, contains_delayed_fields) = StorageLayoutConverter::new(module_storage)
             .type_to_type_layout_with_identifier_mappings(ty)?;
 
-        let (data, bytes_loaded) = {
-            let metadata = module_storage
-                .fetch_existing_module_metadata(
-                    &struct_tag.address,
-                    struct_tag.module.as_ident_str(),
-                )
-                .map_err(|err| err.to_partial())?;
-
-            // If we need to process delayed fields, we pass type layout to remote storage. Remote
-            // storage, in turn ensures that all delayed field values are pre-processed.
-            resource_resolver.get_resource_bytes_with_metadata_and_layout(
+        let metadata = module_storage
```

Comment:
If we have the following calls:
```
exists<A>
borrow<A>
```
this currently creates an entry, and then re-creates it constructing layout again and fetching metadata again. Can we cache these and calculate only once for both instructions, and only update the global value?